libp2p/protocols/connectivity/autonat/service.nim

```diff
@@ -153,7 +154,69 @@ proc schedule(service: AutonatService, switch: Switch, interval: Duration) {.asy
   heartbeat "Scheduling AutonatService run", interval:
     await service.run(switch)
 
+proc handleManualPortForwarding(
+  peerStore: PeerStore,
+  listenAddr: MultiAddress,
+  isIP4: bool): Opt[MultiAddress] =
+  try:
+    let maIP = listenAddr[0]
+    let maWithoutIP = listenAddr[1..^1]
+
+    if maWithoutIP.isErr():
+      return Opt.none(MultiAddress)
+
+    let observedIP =
+      if isIP4:
+        peerStore.getMostObservedIP(IPv4)
+      else:
+        peerStore.getMostObservedIP(IPv6)
+
+    let newMA =
+      if observedIP.isNone() or maIP.get() == observedIP.get():
+        listenAddr
+      else:
+        observedIP.get() & maWithoutIP.get()
+
+    return Opt.some(newMA)
+  except CatchableError as error:
+    debug "Error while handling manual port forwarding", msg = error.msg
+    return Opt.none(MultiAddress)
+
+proc addressMapper(
+  self: AutonatService,
+  peerStore: PeerStore,
+  listenAddrs: seq[MultiAddress]): Future[seq[MultiAddress]] {.gcsafe, async.} =
+
+  var addrs = newSeq[MultiAddress]()
+  for listenAddr in listenAddrs:
+    try:
+      let maIP = listenAddr[0]
+      if maIP.isErr():
+        continue
+      var isIP4 = true
+      let hostIP =
+        if IP4.match(maIP.get()):
+          getBestRoute(initTAddress("8.8.8.8:0")).source
+        elif IP6.match(maIP.get()):
+          isIP4 = false
+          getBestRoute(initTAddress("2600:::0")).source
+        else:
+          continue
+      if not hostIP.isGlobal():
+        if self.networkReachability == NetworkReachability.Reachable:
+          let newMA = handleManualPortForwarding(peerStore, listenAddr, isIP4)
+          if newMA.isSome():
+            addrs.add(newMA.get())
+          continue
```

Comment:
```nim
else:
   # replace with hostIp
```

Reply:
Did something different, let me know what you think.